thavalon-webapp/src/components/Navbar.tsx

```diff
@@ -1,154 +1,50 @@
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import React, {useState} from 'react';
+import React, {MouseEvent} from 'react';
 import {NavLink} from 'react-router-dom';
-import styled from 'styled-components';
 import {faBars} from '@fortawesome/free-solid-svg-icons';
+import "./Navbar.scss";
 
-function openMenu(event: any) {
-    let x = document.getElementById("navbar");
-    if (x === null) {
-        return;
-    }
-    if (x.classList.contains("responsive")) {
-        x.classList.remove("responsive");
-    } else if (x.classList.contains("topnav")) {
-        x.classList.add("responsive");
-    }
-
-    event.preventDefault();
-}
-
-type NavbarProps = {
+interface NavbarProps {
     loggedIn: boolean
+    useMobileMenu: boolean
+    setUseMobileMenu: React.Dispatch<React.SetStateAction<boolean>>
 };
 
-function Navbar(props: NavbarProps) {    
+export function Navbar(props: NavbarProps) {
+    // handle mobile menu functionality
+    let topnavClasses: string = "topnav"
+    if (props.useMobileMenu) {
+        topnavClasses += " responsive";
+    }
+
+    /**
+     * When opening menu, toggle if mobile menu should be displayed or not.
+     * Also suppress click event, in order to avoid changing current page.
+     * 
+     * @param event The event generated by opening the menu.
+     */
+    function openMenu(event: MouseEvent<HTMLAnchorElement>) {
+        props.setUseMobileMenu(!props.useMobileMenu);
+        event.preventDefault();
+    }
+
     return (
-        <NavbarContainer id="navbar" className="topnav">
-            <NavbarItemLeft exact to="/" activeClassName="active" id="homeLink">Home</NavbarItemLeft>
-            <NavbarItemLeft to="/rules" activeClassName="active">Rules</NavbarItemLeft>
+        <div id="navbarContainer" className={topnavClasses}>
+            <NavLink exact to="/" className="navbarItemLeft" activeClassName="active" id="homeLink">Home</NavLink>
+            <NavLink to="/rules" className="navbarItemLeft" activeClassName="active">Rules</NavLink>
             {!props.loggedIn &&
-                <NavbarItemRight to="/login" activeClassName="active">Log In</NavbarItemRight>
+                <NavLink to="/login" className="navbarItemRight" activeClassName="active">Log In</NavLink>
             }
-            <NavbarItemRight to="" className="icon" onClick={openMenu}>
+            <NavLink to="" className="icon navbarItemRight" onClick={openMenu}>

```

Comment:
Does this need to be a `NavLink` if we're not using it for navigation?

Reply:
That seems to be having some issues with highlighting/background colors - I can look more into it, but it's not worth the time.